### PR TITLE
Improved multi-field accessibility and usability.

### PR DESCRIPTION
**Pull request by jonearley**
This PR is a result from a conversation started over in Sufia: https://github.com/projecthydra/sufia/issues/2021#issuecomment-219096513.

### CODE REVIEW

**Comment by mjgiarlo** [comment]
To confirm: @jonearley has a signed CLA, so his work is clear to merge.

File: app/assets/javascripts/hydra-editor/manage_repeating_fields.js [hunk 1]
```diff
@@ -3,139 +3,151 @@
...
+
+        this.options.label = '';
+        var $label = this.element.find("label");
+
+        if ( $label.size ) {
+          var label = $label.data('label') || $.trim($label.contents().filter(function() { return this.nodeType === 3; }).text());
+          this.options.label = ' ' + label;
+        }
+
+        this.controls = $("<span class=\"input-group-btn field-controls\">");
```
Comment (jcoyne) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: This should be passed in from the options.  Did you start from an out of date commit?

**Comment by jcoyne** [comment]
This PR appears to revert the commit: https://github.com/projecthydra/hydra-editor/commit/be2ad3b7cdbba4fcb23ee78e15ab4b7f2ae0d471

I was rather fond of that commit, so can you work it back in?

**Comment by jcoyne** [comment]
Here's a few other things to take a look at:
1. Please take a look at the Jasmine tests, they appear to need some adjustment.
1. Please update your commit to explain what you did to improve accessibility.
1. Please add tests for the features you have added.

**Comment by jonearley** [comment]
@jcoyne I've added back in the be2ad3b commit (mostly). Since the label is added into the add and remove buttons I've swapped out the html option to a text option as a compromise.

**Comment by jcoyne** [comment]
@jonearley still think you should make the html a template that goes into the options.  It could be function that takes the text as an argument, or you could use something like [Handlebars](http://handlebarsjs.com/)

**Comment by jonearley** [comment]
@jcoyne I've updated the PR so the add and remove buttons can be passed in as a html option. I added a span to select on and insert text. It works, but maybe there is a better way.

File: app/assets/javascripts/hydra-editor/manage_repeating_fields.js [hunk 2]
```diff
@@ -3,139 +3,171 @@
...
+        if ($label.size && options.labelControls) {
+          var label = $label.data('label') || $.trim($label.contents().filter(function() { return this.nodeType === 3; }).text());
+          this.options.label = ' ' + label;
+        }
+
+        var $addHtml = $(options.addHtml);
+        $addHtml.find('.controls-add-text').html(options.addText + this.options.label);
+        this.adder    = $addHtml;
+
+        var $removeHtml = $(options.removeHtml);
```
Comment (jcoyne) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: please encapsulate this (and the following 2 lines) into a function.

File: app/assets/javascripts/hydra-editor/manage_repeating_fields.js [hunk 3]
```diff
@@ -3,139 +3,171 @@
...
-              this.element.bind('managed_field:remove', this.options.remove);
-          },
-
-          addToList: function( event ) {
+    var FieldManager = function (element, options) {
+        this.element = $(element);
+        this.options = options;
+
+        this.options.label = '';
+        var $label = this.element.find("label");
```
Comment (jcoyne) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: Please move lines 11-16 into a function.

File: app/assets/javascripts/hydra-editor/manage_repeating_fields.js [hunk 4]
```diff
@@ -3,139 +3,171 @@
...
-            field.remove();
+            this._manageFocus()
+        },
+
+        inputIsEmpty: function($activeField) {
+            return $activeField.children('input.multi-text-field').val() === '';
+        },
+
+        _newField: function ($activeField) {
+            var $newField = this.createNewField($activeField);
```
Comment (jcoyne) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: Is there any reason to assign to a variable, or can you just return the result of the function call?

Comment (jonearley) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: Good catch, I pushed a change to fix this.

File: app/assets/javascripts/hydra-editor/manage_repeating_fields.js (hunk 3 above)
Comment (jonearley) [inline app/assets/javascripts/hydra-editor/manage_repeating_fields.js]: Good idea. My recent force push reflects your suggestion.

**Pull request merged**